packages/labs/react/src/create-component.ts

```diff
@@ -31,6 +31,18 @@ export type WebComponentProps<I extends HTMLElement> = React.DetailedHTMLProps<
 > &
   ElementProps<I>;
 
+// Type below taken from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/05094b6df4fb6f4404333725f13d63dd8aabeb34/types/react/index.d.ts#L825-L830
+// We make a copy here so that preact users can still use it as it is not
+// included in preact/compat yet.
+// See: https://github.com/preactjs/preact/issues/4124
+/** Ensures that the props do not include ref at all */
+type PropsWithoutRef<P> =
+  // Omit would not be sufficient for this. We'd like to avoid unnecessary mapping and need a distributive conditional to support unions.
+  // see: https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#distributive-conditional-types
+  // https://github.com/Microsoft/TypeScript/issues/28339
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  P extends any ? ('ref' extends keyof P ? Omit<P, 'ref'> : P) : P;
+
```

Comment:
spoke offline – would be nice to isolate this in a separate file and include the license.

Reply:
Thank you, I have split out `PropsWithoutRef` into its own file and included all license and attributions there.